app/src/test/java/mozilla/lockbox/mocks/MockLoginsStorage.kt

```diff
@@ -43,6 +46,30 @@ open class MockLoginsStorage : LoginsStorage {
             SyncResult.fromValue(true)
         }
     }
+    override fun add(login: ServerPassword): SyncResult<String> {
+        return list().then {
+            if (login.id == "") {
+                SyncResult.fromException<String>(InvalidRecordException("record id missing"))
+            } else if (null != it.find { it.id == login.id }) {
+                SyncResult.fromException<String>(IdCollisionException("record already exists"))
+            }
+
+            all.add(login)
+            SyncResult.fromValue(login.id)
+        }
+    }
+
+    override fun update(login: ServerPassword): SyncResult<Unit> {
+        return list().then {
+            val position = it.indexOfFirst { it.id == login.id }
+            if (position == -1) {
```

Comment:
Nit: prefer 

a) putting the positive branch first and the exceptional branch second.
b) using the numbers `0`, `1` and `n`.

i.e.

```kotlin
if (position >= 0) {

} else {
    throw …
}
```